alma/lib/Services/InsuranceApiService.php

```diff
@@ -159,9 +156,48 @@ public function subscribeInsurance($subscriptionData, $idTransaction)
                     $idTransaction
                 )
             );
-
         }
 
         throw new InsuranceSubscriptionException();
     }
+
+    /**
+     * @param $insuranceContracts
+     * @param $cartId
+     *
+     * @return false|void
+     */
+    public function sendCmsReferenceSubscribedForTracking($insuranceContracts, $cartId)
+    {
+        $cmsReferences = [];
+
+        if (empty($insuranceContracts)) {
+            Logger::instance()->warning(
+                sprintf(
+                    '[Alma] No insurance contract to send, cartId: "%s"',
+                    $cartId
+                )
+            );
+
+            return false;
+        }
+
+        foreach ($insuranceContracts as $insuranceContract) {
+            $cmsReferences[] = $insuranceContract['cms_reference'];
+        }
+
+        try {
+            $this->almaApiClient->insurance->sendCustomerCart($cmsReferences, $cartId);
+        } catch (RequestError $e) {
+            Logger::instance()->error(
+                sprintf(
+                    '[Alma] Error while sending the cms_reference for tracking, message "%s", trace "%s", cmsReference : "%s", cartId: "%s"',
+                    $e->getMessage(),
+                    $e->getTraceAsString(),
+                    json_encode($cmsReferences),
+                    $cartId
+                )
+            );
+        }
+    }
```

Comment:
I need explanation please

Reply:
@Francois-Gomis 
We fixed the csmReferences returned for tracking
@CamilleFljt